modules/core/src/main/java/io/fluo/core/metrics/MetricNames.java

```diff
@@ -14,65 +14,81 @@
 
 package io.fluo.core.metrics;
 
+import com.google.common.base.Preconditions;
 import io.fluo.api.config.FluoConfiguration;
 
 public class MetricNames {
 
-  private final String txLockWait;
-  private final String txTime;
+  private final String txLockWaitTime;
+  private final String txExecTime;
+  private final String txWithCollision;
   private final String txCollisions;
-  private final String txSet;
-  private final String txRead;
+  private final String txEntriesSet;
+  private final String txEntriesRead;
   private final String txLocksTimedOut;
   private final String txLocksDead;
   private final String txStatus;
 
   private final String notificationsQueued;
 
+  private final String oracleResponseTime;
   private final String oracleClientStamps;
-  private final String oracleClientRpc;
   private final String oracleServerStamps;
 
-
   public static final String METRICS_ID_PROP = FluoConfiguration.FLUO_PREFIX + ".metrics.id";
 
-  public MetricNames(String hostId) {
-    txLockWait = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.lockWait.";
-    txTime = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.time.";
-    txCollisions = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.collisions.";
-    txSet = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.set.";
-    txRead = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.read.";
-    txLocksTimedOut = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.locks.timedout.";
-    txLocksDead = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.locks.dead.";
-    txStatus = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".tx.status.";
-
-    notificationsQueued =
-        FluoConfiguration.FLUO_PREFIX + "." + hostId + ".worker.notifications.queued";
-
-    oracleClientStamps = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".oracle.client.stamps";
-    oracleClientRpc =
-        FluoConfiguration.FLUO_PREFIX + "." + hostId + ".oracle.client.rpc.getStamps.time";
-    oracleServerStamps = FluoConfiguration.FLUO_PREFIX + "." + hostId + ".oracle.server.stamps";
+  public MetricNames(String hostId, String appName) {
+    Preconditions.checkArgument(!appName.contains("."), "Fluo App name should not contain '.': "
+        + appName);
+    Preconditions.checkArgument(!hostId.contains("."), "Host ID should not contain '.': " + hostId);
```

Comment:
I was wondering if FluoConfiguration should also do this check when setting the app name.  I checked and it already does for some reason.